lib/core/ingestion/ingester.rb

```diff
@@ -0,0 +1,81 @@
+#
+# Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+# or more contributor license agreements. Licensed under the Elastic License;
+# you may not use this file except in compliance with the Elastic License.
+#
+
+# frozen_string_literal: true
+
+require 'utility/logger'
+
+module Core
+  module Ingestion
+    class Ingester
+      def initialize(sink_strategy)
+        @sink_strategy = sink_strategy
+        @ingested_count = 0
+        @ingested_volume = 0
+        @deleted_count = 0
+      end
+
+      def ingest(document)
+        unless document&.any?
+          Utility::Logger.warn('Connector attempted to ingest an empty document, skipping')
+          return
+        end
+
+        serialized_document = @sink_strategy.serialize(document)
+        @sink_strategy.ingest(document['id'], serialized_document)
+
+        @ingested_count += 1
+        @ingested_volume += serialized_document.bytesize
```

Comment:
🤔 You add another layer `Ingester` above `Sink`. 

I think one of the benefits of having ingestion stats in Sink, is that we could have more accurate data. E.g. we could increment @ingested_count when flush is called. But I see you serialize everything (operation, payload) in the queue, so it's not easy to count the docs and volume anyway.

Reply:
I see, I've overlooked this, I'll address in a separate PR